src/server/protocol.ts

```diff
@@ -2383,8 +2363,9 @@ export interface GeterrRequestArgs {
     /**
      * List of file names for which to compute compiler errors.
      * The files will be checked in list order.
+     * Files with ranges specified will be checked first.
      */
-    files: string[];
+    files: (string | FileRangesRequestArgs)[];

```

Comment:
` GeterrForProjectRequest` does this also need to enable region based .. we already create high priority list etc in there.

Reply:
I wasn't planning on enabling region-based diagnostics for the project diagnostics scenario at the moment. From my last talk with Matt on the vscode usage of this command, it's still marked as experimental on the vscode side and it has known issues, so I figured I wouldn't support it at first.